pkg/tsdb/phlare/kinds/dataquery/types_dataquery_gen.go

```diff
@@ -0,0 +1,48 @@
+// Code generated - EDITING IS FUTILE. DO NOT EDIT.
+//
+// Generated by:
+//     public/app/plugins/gen.go
+// Using jennies:
+//     PluginGoTypesJenny
+//
+// Run 'make gen-cue' from repository root to regenerate.
+
+package dataquery
+
+// Defines values for QueryType.
+const (
+	QueryTypeBoth QueryType = "both"
+
+	QueryTypeMetrics QueryType = "metrics"
+
+	QueryTypeProfile QueryType = "profile"
+)
+
+// PhlareDataQuery defines model for PhlareDataQuery.
+type PhlareDataQuery struct {
+	// For mixed data sources the selected datasource is on the query level.
+	// For non mixed scenarios this is undefined.
+	// TODO find a better way to do this ^ that's friendly to schema
+	// TODO this shouldn't be unknown but DataSourceRef | null
+	Datasource *interface{} `json:"datasource,omitempty"`
+	GroupBy    []string     `json:"groupBy"`
+
+	// true if query is disabled (ie should not be returned to the dashboard)
+	Hide *bool `json:"hide,omitempty"`
+
+	// Unique, guid like, string used in explore mode
+	Key           *string `json:"key,omitempty"`
+	LabelSelector string  `json:"labelSelector"`
+	ProfileTypeId string  `json:"profileTypeId"`
+
+	// Specify the query flavor
+	// TODO make this required and give it a default
+	QueryType QueryType `json:"queryType"`
```

Comment:
I am not sure the Go typing is correct. The request currently is like this:
```
type DataQuery struct {
	// RefID is the unique identifier of the query, set by the frontend call.
	RefID string

	// QueryType is an optional identifier for the type of query.
	// It can be used to distinguish different types of queries.
	QueryType string

	// MaxDataPoints is the maximum number of datapoints that should be returned from a time series query.
	MaxDataPoints int64

	// Interval is the suggested duration between time points in a time series query.
	Interval time.Duration

	// TimeRange is the Start and End of the query as sent by the frontend.
	TimeRange TimeRange

	// JSON is the raw JSON query and includes the above properties as well as custom properties.
	JSON json.RawMessage
}
```
And so things like `ProfileTypeId` are inside a JSON property and so not on the same level as `QueryType` for example so not sure the code will work with this type.